src/commands/pressable-grant-access.php

```diff
@@ -286,6 +290,19 @@ private function search_for_site_url( string $site_url ): ?\stdClass {
 		return null;
 	}
 
+	/**
+	 * Search for a site by ID
+	 */
+	private function get_site_by_id( int $site_id ): ?\stdClass {
+		$site = $this->api_helper->call_pressable_api( "sites/$site_id", 'GET', array() );
+		// If we have no results, return an empty null.
+		if ( is_null( $site ) || empty( $site->data ) ) {
```

Comment:
Using `empty()` will cover null states also, so no need for the `is_null()` check!

Reply:
Ah, yes. I understand your thinking there. A few things to consider:

* The `call_pressable_api()` function could return a JSON encoded error message (or some other unexpected non-null value JSON encoded data). Null is returned if `json_encode()` fails to encode what's passed to it, which is a limited case to cover and if it evaluates to true, it doesn't tell us `$site` is anything we can actually use (or even if it's an object).
* Internally, `empty()` does an `isset()` first (on the object property in this case), so all you need is `empty( $site->data )` to safely evaluate whether `$site->data` contains data that doesn't evaluate to empty.

Reply:
> Internally, empty() does an isset() first (on the object property in this case)

Well, that's a big TIL for me! I'll fix that here and in `pressable_sites()`